source/baseObject.py

```diff
@@ -122,8 +122,11 @@ class AutoPropertyObject(with_metaclass(AutoPropertyType, object)):
 	"""
 
 	#: Tracks the instances of this class; used by L{invalidateCaches}.
-	#: @type: weakref.WeakKeyDictionary
-	__instances=weakref.WeakKeyDictionary()
+	#: @type: weakref.WeakValueDictionary
+	# Theoretically we could use weakref.WeakSet, 
+	# but as there is no easy way to have NVDAObjects and TextInfos remain hashable in Python3, as they override equality,
+	# We store them on a WeakValueDictionary, keyed by their id (address).

```

Comment:
This comment suggest that we could theoretically use a WeakSet, but that's not going to work as in sets, only hashable types are allowed. Doesn't that mean we theoretically can not use a set?